core/src/avm1/property_decl.rs

```diff
@@ -126,66 +122,18 @@ macro_rules! declare_properties {
         crate::avm1::property_decl::Declaration {
             name: $name,
             kind: declare_properties!(@__kind $kind ($($args),*)),
-            attributes: 0,
+            attributes: crate::avm1::property::Attribute::empty(),
         }
     };
-    (@__prop $kind:ident($name:literal $(,$args:expr)*; $($attributes:ident)|*$(; version($version:tt))?) ) => {
+    (@__prop $kind:ident($name:literal $(,$args:expr)*; $($attributes:ident)|*) ) => {
         crate::avm1::property_decl::Declaration {
             name: $name,
             kind: declare_properties!(@__kind $kind ($($args),*)),
-            /*
-                WARNING: HORRIBLE HACK AHEAD!
-
-                To declare property attributes in a way that is valid in `const` context,
-                we store them as raw `u8`s and do the bitflag management ourselves.
-
-                Here are two better ways that unfortunately don't work.
-
-                A) `Attribute::FOO | Attribute::BAR`
-
-                This can't be used, because operator overloading doesn't work in `const` context.
-
-                B) `Attributes::from_bits_truncate(Attribute::FOO.bits() | Attribute::BAR.bits())`
-
-                Here, everything is a proper `const fn` and so this should work correctly,
-                but NO!, we hit an ICE in the compiler :(
-                See:
-                    https://github.com/rust-lang/rust/issues/81899
-                    https://github.com/rust-lang/rust/issues/84957
-
-                TODO: use the `B)` desugaring once the above ICE is fixed.
-            */
-            attributes: 0 $(| declare_properties!(@__attr $attributes))* $(| declare_properties!(@__version $version))*,
+            attributes: crate::avm1::property::Attribute::from_bits_truncate(
```

Comment:
nit: Cannot this be simplified to the following?
```rs
attributes: crate::avm1::property::Attribute::empty() $(| crate::avm1::property::Attribute::$attributes)*,
```

Reply:
It can't, because overloaded operators aren't available in `const` contexts.